pkg/controller/jobs/jobset/jobset_webhook.go

```diff
@@ -57,9 +67,41 @@ func (w *JobSetWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	log.V(5).Info("Applying defaults", "jobset", klog.KObj(jobSet))
 
 	jobframework.ApplyDefaultForSuspend(jobSet, w.manageJobsWithoutQueueName)
+
+	if canDefaultManagedBy(jobSet.Spec.ManagedBy) {
+		localQueueName, found := jobSet.Labels[constants.QueueLabel]
+		if !found {
+			return nil
+		}
+		clusterQueueName, err := w.queues.ClusterQueueFromLocalQueue(queue.QueueKey(jobSet.ObjectMeta.Namespace, localQueueName))
+		if err != nil {
+			return err
+		}
+		clusterQueue, found := w.cache.GetClusterQueue(clusterQueueName)
+		if !found {
+			return nil
+		}
+		for admissionCheckName := range clusterQueue.AdmissionChecks {
+			admissionCheck, ok := w.cache.GetAdmissionCheck(admissionCheckName)
```

Comment:
this is taking multiple locks. Can we have a single function `AdmissionChecksForClusterQueue`?